docs(python): Document how to migrate custom `Hub` usage (#10521)
getsentry/sentry-docs: docs/platforms/python/migration/1.x-to-2.x.mdx

The old APIs will continue to work in 2.0, but we recommend migrating to their new counterparts as soon as possible since they will be removed in the next major release.

### Activating a Custom Hub

> If you are using a custom hub to isolate event data, you can instead use a custom isolation scope. To activate the custom isolation scope, you will need to explicitly pass the isolation scope to the `use_isolation_scope`, which creates a context manager that activates the scope. Here is an example of how to change your code to make it work:

Reviewer: ```suggestion
If you were using a custom hub to isolate event data, we recommend using a custom isolation scope instead. To activate the custom isolation scope, you will need to explicitly pass the isolation scope to the `use_isolation_scope`, which creates a context manager that activates the scope. Here is an example of how to change your code to make it work:
```
